crates/node/src/lib.rs

```diff
@@ -396,22 +391,18 @@ impl Node {
                 .ctx_manager
                 .get_own_signing_key(&context.id, &our_identity)?;
 
-            let shared_key = SharedKey::new(
-                &our_sending_key,
-                &VerifyingKey::from_bytes(&executor_public_key)?,
-            );
+            let shared_key = SharedKey::from_sk(&our_sending_key);
 
             let artifact_encrypted = shared_key
                 .encrypt(outcome.artifact.clone(), [0; aead::NONCE_LEN])
                 .unwrap();
 
             let message = to_vec(&BroadcastMessage::StateDelta {
-                // We are encrypting only the artifact so we can use the context_id and own_id for decrypting
+                // We are encrypting only the artifact so we can use the context_id decrypting
```

Comment:
but we're not, right? not using the context id to decrypt

Reply:
Yes, the comment should be better worded as we use the `context_id` so we can fetch the `SenderKey`